### PR TITLE
Send meta options from backend to frontend and back to backend

### PR DESCRIPTION
**Pull request by joelbarmettlerUZH**

### CODE REVIEW

File: Simulator/Owner/SpaceOwners/StationaryOwner.py [hunk 1]
```diff
@@ -15,6 +15,15 @@ class StationaryOwner(SpaceOwner):
...
     positions = ">0"
+    meta = [
+        {
+            "key": "size",
+            "label": "Field Size",
+            "description": "Size of Reserved Field (qubic)",
+            "type": "int",
+            "value": random.randint(0, 100)
+        }
+    ]
```
Comment (thommann) [inline Simulator/Owner/SpaceOwners/StationaryOwner.py]: no x, y, z values?

File: Bidding/BiddingStationaryOwner.py [hunk 2]
```diff
@@ -14,6 +14,15 @@
...
     description = "A bidding owner interested in a set of cubes"
+    meta = [
+        {
+            "key": "size",
+            "label": "Field Size",
+            "description": "Size of Reserved Field (qubic)",
+            "type": "int",
+            "value": random.randint(0, 100)
+        }
+    ]
```
Comment (thommann) [inline Bidding/BiddingStationaryOwner.py]: same

File: Visualizer/src/API/typedefs.js [hunk 3]
```diff
@@ -146,6 +146,15 @@
  * @property {int} total_number_of_reallocations
  */
 
+/**
+ * @typedef {Object} RawMeta
+ * @property {string} key
+ * @property {string} name
+ * @property {string} description
+ * @property {string} type
+ * @property {string|number|boolean} default
```
Comment (thommann) [inline Visualizer/src/API/typedefs.js]: it's called value not default in the backend

File: Visualizer/src/components/home/owner/OwnerForm.vue [hunk 4]
```diff
@@ -30,11 +30,16 @@ const simulationConfig = useSimulationConfigStore();
...
+
 /**
  * Whenever the selected ownerType changes, make sure the requirements for minimum
  * and maximum number of locations are met
  */
 const updateLocationsForOwner = () => {
+  owner.value.meta = ownerProperties.value.meta;
   if (owner.value.locations > owner.value.maxLocations) {
     owner.value.locations = owner.value.locations.slice(0, owner.value.maxLocations);
   }
```
Comment (thommann) [inline Visualizer/src/components/home/owner/OwnerForm.vue]: why is the `ownerProperties` variable needed?

File: Visualizer/src/stores/simulation.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { defineStore } from "pinia";
 import { useStorage } from "@vueuse/core";
-import { emitAgentsSelectedEvent, emitTickEvent } from "../scripts/emitter";
+import { emitAgentsSelectedEvent, emitTickEvent } from "@/scripts/emitter";
```
Comment (thommann) [inline Visualizer/src/stores/simulation.js:3]: use relative paths

**Review by thommann — CHANGES_REQUESTED** [review]

File: Visualizer/src/components/home/owner/OwnerForm.vue (hunk 4 above)
Comment (joelbarmettlerUZH) [reply Visualizer/src/components/home/owner/OwnerForm.vue]: To overwrite the meta properties of a created owner with the meta properties of the available owner type whenever the type changes

File: Visualizer/src/stores/simulation.js (hunk 5 above)
Comment (joelbarmettlerUZH) [reply Visualizer/src/stores/simulation.js:3]: WebStorm 2022.2.1RC always changes it back to absolute paths. I think it does not really matter, but we should freeze it in the linting rules.

File: Visualizer/src/components/home/owner/OwnerForm.vue (hunk 4 above)
Comment (thommann) [reply Visualizer/src/components/home/owner/OwnerForm.vue]: shouldn't then the maxLocations be overwritten as well?

Comment (joelbarmettlerUZH) [reply Visualizer/src/components/home/owner/OwnerForm.vue]: true true true

**Review by thommann — APPROVED** [review]

**Pull request merged**